lib/modification/hooks/use-warn-if-unsaved-changes.ts

```diff
@@ -0,0 +1,37 @@
+import {useRouter} from 'next/router'
+import {useEffect} from 'react'
+
+/**
+ * Prevent navigation when changes have not been saved.
+ */
+export default function useWarnIfUnsavedChanges(unsavedChanges: boolean) {
+  const router = useRouter()
+  const message =
+    'Please save or discard your changes before leaving this page.'
```

Comment:
UI doesn't give a clear indication of how to "discard" changes, so maybe

```suggestion
    'Please save or undo your changes before leaving this page.'
```